src/main/java/facade/UserFacade.java

```diff
@@ -29,7 +55,14 @@
    * @throws InvalidUserCookieException if the cookie is invalid.
    */
   public User getUser(String username, String cookie) throws UserNotFoundException, InvalidUserCookieException {
-    return null;
+      LOGGER.info("UserFacade.getUser");
+      if(!validateUserCookie(cookie)){
```

Comment:
What if we just had the `validateUserCookie` function throw the exception. That would reduce code duplication.